packages/aws-cdk-lib/aws-iam/lib/oidc-provider-native.ts

```diff
@@ -0,0 +1,179 @@
+import { Construct } from 'constructs';
+import { CfnOIDCProvider } from './iam.generated';
+import { Arn, IResource, Resource, Token } from '../../core';
+
+/**
+ * Represents an IAM OpenID Connect provider.
+ *
+ */
+export interface IOidcProvider extends IResource {
+  /**
+   * The Amazon Resource Name (ARN) of the IAM OpenID Connect provider.
+   *
+   * @attribute
+   */
+  readonly oidcProviderArn: string;
+
+  /**
+   * The issuer for OIDC Provider
+   *
+   * @attribute
+   */
+  readonly oidcProviderIssuer: string;
+}
```

Comment:
I think we don't actually need a new interface here since this is identical to the old `IOpenIdConnectProvider` https://github.com/aws/aws-cdk/blob/d1b3c817676df87d24ad1b97af9b304f7c8526a3/packages/aws-cdk-lib/aws-iam/lib/oidc-provider.ts#L13-L27

Let's just implement that one instead of creating a new duplicate.